code/modules/clothing/suits/reactive_armour.dm

```diff
@@ -150,70 +221,94 @@
+		thrown_items[A] = A
+
+	reactivearmor_cooldown = world.time + reactivearmor_cooldown_duration
+	return FALSE
 
 /obj/item/clothing/suit/armor/reactive/table
 	name = "reactive table armor"
 	desc = "If you can't beat the memes, embrace them."
+	cooldown_message = "<span class='danger'>The reactive table armor's fabricators are still on cooldown!</span>"
+	emp_message = "<span class='danger'>The reactive table armor's fabricators click and whirr ominously for a moment...</span>"
 	var/tele_range = 10
 
-/obj/item/clothing/suit/armor/reactive/table/hit_reaction(mob/living/carbon/human/owner, atom/movable/hitby, attack_text = "the attack", final_block_chance = 0, damage = 0, attack_type = MELEE_ATTACK)
-	if(!active)
-		return FALSE
-	if(prob(hit_reaction_chance))
-		var/mob/living/carbon/human/H = owner
-		if(world.time < reactivearmor_cooldown)
-			owner.visible_message("<span class='danger'>The reactive table armor's fabricators are still on cooldown!</span>")
-			return FALSE
-		owner.visible_message("<span class='danger'>The reactive teleport system flings [H] clear of [attack_text] and slams [H.p_them()] into a fabricated table!</span>")
-		owner.visible_message("<font color='red' size='3'>[H] GOES ON THE TABLE!!!</font>")
-		owner.Paralyze(40)
-		do_teleport(owner, get_turf(owner), tele_range, no_effects = TRUE, channel = TELEPORT_CHANNEL_BLUESPACE)
-		new /obj/structure/table(get_turf(owner))
-		reactivearmor_cooldown = world.time + reactivearmor_cooldown_duration
-		return TRUE
-	return FALSE
+/obj/item/clothing/suit/armor/reactive/table/reactive_activation(mob/living/carbon/human/owner, atom/movable/hitby, attack_text = "the attack", final_block_chance = 0, damage = 0, attack_type = MELEE_ATTACK)
+	owner.visible_message("<span class='danger'>The reactive teleport system flings [owner] clear of [attack_text] and slams [owner.p_them()] into a fabricated table!</span>")
+	owner.visible_message("<font color='red' size='3'>[owner] GOES ON THE TABLE!!!</font>")
+	owner.Knockdown(30)
+	owner.apply_damage(10, BRUTE)
+	owner.apply_damage(40, STAMINA)
+	playsound(owner, 'sound/effects/tableslam.ogg', 90, TRUE)
+	SEND_SIGNAL(owner, COMSIG_ADD_MOOD_EVENT, "table", /datum/mood_event/table)
+	do_teleport(owner, get_turf(owner), tele_range, no_effects = TRUE, channel = TELEPORT_CHANNEL_BLUESPACE)
+	new /obj/structure/table(get_turf(owner))
+	reactivearmor_cooldown = world.time + reactivearmor_cooldown_duration
+	return TRUE
+
+/obj/item/clothing/suit/armor/reactive/table/emp_activation(mob/living/carbon/human/owner, atom/movable/hitby, attack_text = "the attack", final_block_chance = 0, damage = 0, attack_type = MELEE_ATTACK)
+	owner.visible_message("<span class='danger'>The reactive teleport system flings [owner] clear of [attack_text] and slams [owner.p_them()] into a fabricated glass table!</span>")
+	owner.visible_message("<font color='red' size='3'>[owner] GOES ON THE GLASS TABLE!!!</font>")
```

Comment:
I forget, does tableslamming someone into a glass table just do the normal glass table shattering effect/stun, or does it also do the brute damage, knockdown, and stamina damage from normal tableslamming? if the latter, this should also apply the same damage and knockdown amounts that the normal activation does (alongside the normal effects for moving onto the same tile as a glass table).